frontend/location.py

```diff
@@ -394,6 +395,14 @@ def LeftTokenForCompoundWord(w):
         return None
 
 
+def LeftTokenForRedirOp(w):
+    # type: (RedirOp) -> Optional[Token]
```

Comment:
I think this should just be Token, not Optional[Token]

(unfortunately we don't have MyPy configured to strictly enforce this)